waiter/src/waiter/state.clj

```diff
@@ -46,6 +53,37 @@
   (and (zero? slots-used)
        (not-any? #(contains? status-tags %) [:killed :locked])))
 
+(defn- find-max
+  "Uses the provided comparison function to find the max element in collection."
+  [comparison-fn collection]
+  (reduce #(if (pos? (comparison-fn %1 %2)) %1 %2) (first collection) (rest collection)))
+
+(defn- only-lingering-requests?
+  "Retrurns truthy value if the instance is idle or processing expired requests based on the earliest-request-threshold-time"
+  [request-id->use-reason-map earliest-request-threshold-time]
+  (every? #(t/before? % earliest-request-threshold-time)
+          (->> request-id->use-reason-map
+               vals
+               (map :time))))
+
+(defn find-oldest-acceptable-expired-instance
```

Comment:
This function feels a lot like `find-killable-instance`.  Could we merge the logic?  See my other comments for more color.